tests/jvm/src/test/resources/example/Types.scala

```diff
@@ -56,7 +56,7 @@ object Test/*<=types.Test.*/ {
     val compoundType2/*<=types.Test.C#compoundType2.*/: M/*=>types.Test.M#*/ with N/*=>types.Test.N#*/ = ???/*=>scala.Predef.`???`().*/
     val compoundType3/*<=types.Test.C#compoundType3.*/: M/*=>types.Test.M#*/ with N/*=>types.Test.N#*/ { def k/*<=local1*/: Int/*=>scala.Int#*/ } = ???/*=>scala.Predef.`???`().*/
     val compoundType4/*<=types.Test.C#compoundType4.*/ = new /*<=local2*/{ def k/*<=local3*/: Int/*=>scala.Int#*/ = ???/*=>scala.Predef.`???`().*/ }
-    val compoundType5/*<=types.Test.C#compoundType5.*/ = new M/*=>types.Test.M#*/ with N/*=>types.Test.N#*/
+    val compoundType5/*<=types.Test.C#compoundType5.*/ = new /*<=local4*/M/*=>types.Test.M#*/ with N/*=>types.Test.N#*/
```

Comment:
```suggestion
    val compoundType5/*<=types.Test.C#compoundType5.*/ = new M/*=>types.Test.M#*/ with N/*=>types.Test.N#*/
```
This should not be added, we might need to filter it out in the `g.Constant` match case.

Reply:
Looks like this is irrelevant to this change 🤔 
I'm still not sure when there's symbol occurrence for `new` literal, but I just added the following two lines of code (in the latest `main` branch) to `Types.scala` and run `++2.12.14` -> `save-expect`.
```scala
    final val clazzOfJStr = classOf[java.lang.String]
    final val clazzOfM = classOf[M]
```

And I still get

```scala
    val compoundType5/*<=types.Test.C#compoundType5.*/ = new /*<=local4*/M/*=>types.Test.M#*/ with N/*=>types.Test.N#*/
```

Maybe we should figure out what's going on and fix it in another issue/PR?

<details>
<summary>diff</summary>

```diff
diff --git a/semanticdb/integration/src/main/scala/example/Types.scala b/semanticdb/integration/src/main/scala/example/Types.scala
index be2826873..d63a65cca 100644
--- a/semanticdb/integration/src/main/scala/example/Types.scala
+++ b/semanticdb/integration/src/main/scala/example/Types.scala
@@ -114,5 +114,7 @@ object Test {
     final val clazzOfInt = classOf[Int]
     final val clazzOfOption = classOf[Option[Int]]
 
+    final val clazzOfJStr = classOf[java.lang.String]
+    final val clazzOfM = classOf[M]
   }
 }
diff --git a/tests/jvm/src/test/resources/example/Types.scala b/tests/jvm/src/test/resources/example/Types.scala
index 1ebeb095a..5fe4bae11 100644
--- a/tests/jvm/src/test/resources/example/Types.scala
+++ b/tests/jvm/src/test/resources/example/Types.scala
@@ -56,7 +56,7 @@ object Test/*<=types.Test.*/ {
     val compoundType2/*<=types.Test.C#compoundType2.*/: M/*=>types.Test.M#*/ with N/*=>types.Test.N#*/ = ???/*=>scala.Predef.`???`().*/
     val compoundType3/*<=types.Test.C#compoundType3.*/: M/*=>types.Test.M#*/ with N/*=>types.Test.N#*/ { def k/*<=local1*/: Int/*=>scala.Int#*/ } = ???/*=>scala.Predef.`???`().*/
     val compoundType4/*<=types.Test.C#compoundType4.*/ = new /*<=local2*/{ def k/*<=local3*/: Int/*=>scala.Int#*/ = ???/*=>scala.Predef.`???`().*/ }
-    val compoundType5/*<=types.Test.C#compoundType5.*/ = new M/*=>types.Test.M#*/ with N/*=>types.Test.N#*/
+    val compoundType5/*<=types.Test.C#compoundType5.*/ = new /*<=local4*/M/*=>types.Test.M#*/ with N/*=>types.Test.N#*/
     val compoundType6/*<=types.Test.C#compoundType6.*/ = new /*<=local5*/M/*=>types.Test.M#*/ with N/*=>types.Test.N#*/ { def k/*<=local6*/: Int/*=>scala.Int#*/ = ???/*=>scala.Predef.`???`().*/ }
 
     val annType1/*<=types.Test.C#annType1.*/: T/*=>types.T#*/ @ann/*=>types.ann#*/(42) = ???/*=>scala.Predef.`???`().*/
@@ -114,5 +114,7 @@ object Test/*<=types.Test.*/ {
     final val clazzOfInt/*<=types.Test.Literal.clazzOfInt.*/ = classOf/*=>scala.Predef.classOf().*/[Int]
     final val clazzOfOption/*<=types.Test.Literal.clazzOfOption.*/ = classOf/*=>scala.Predef.classOf().*/[Option[Int]]
 
+    final val clazzOfJStr/*<=types.Test.Literal.clazzOfJStr.*/ = classOf/*=>scala.Predef.classOf().*/[java.lang.String]
+    final val clazzOfM/*<=types.Test.Literal.clazzOfM.*/ = classOf/*=>scala.Predef.classOf().*/[M]
   }
 }
diff --git a/tests/jvm/src/test/resources/metac.expect b/tests/jvm/src/test/resources/metac.expect
index d957bd119..fc1309faa 100644
--- a/tests/jvm/src/test/resources/metac.expect
+++ b/tests/jvm/src/test/resources/metac.expect
@@ -4053,8 +4053,8 @@ Schema => SemanticDB v4
 Uri => semanticdb/integration/src/main/scala/example/Types.scala
 Text => non-empty
 Language => Scala
-Symbols => 140 entries
-Occurrences => 252 entries
+Symbols => 142 entries
+Occurrences => 256 entries
 
 Symbols:
 local0 => abstract method k: Int
@@ -4322,13 +4322,19 @@ types/Test.C#typeRef4. => val method typeRef4: List[Int]
 types/Test.C#x. => val method x: p.X
   p => types/Test.C#p.
   X => types/P#X#
-types/Test.Literal. => final object Literal extends AnyRef { +12 decls }
+types/Test.Literal. => final object Literal extends AnyRef { +14 decls }
   AnyRef => scala/AnyRef#
 types/Test.Literal.bool. => final val method bool: true
 types/Test.Literal.char. => final val method char: 'a'
 types/Test.Literal.clazzOfInt. => final val method clazzOfInt: Class[Int]
   Class => java/lang/Class#
   Int => scala/Int#
+types/Test.Literal.clazzOfJStr. => final val method clazzOfJStr: Class[String]
+  Class => java/lang/Class#
+  String => java/lang/String#
+types/Test.Literal.clazzOfM. => final val method clazzOfM: Class[M]
+  Class => java/lang/Class#
+  M => types/Test.M#
 types/Test.Literal.clazzOfOption. => final val method clazzOfOption: Class[Option[Int]]
   Class => java/lang/Class#
   Option => scala/Option#
@@ -4626,6 +4632,10 @@ Occurrences:
 [113:27..113:34): classOf => scala/Predef.classOf().
 [114:14..114:27): clazzOfOption <= types/Test.Literal.clazzOfOption.
 [114:30..114:37): classOf => scala/Predef.classOf().
+[116:14..116:25): clazzOfJStr <= types/Test.Literal.clazzOfJStr.
+[116:28..116:35): classOf => scala/Predef.classOf().
+[117:14..117:22): clazzOfM <= types/Test.Literal.clazzOfM.
+[117:25..117:32): classOf => scala/Predef.classOf().
 
 semanticdb/integration/src/main/scala/example/ValPattern.scala
 --------------------------------------------------------------
diff --git a/tests/jvm/src/test/resources/metacp.expect b/tests/jvm/src/test/resources/metacp.expect
index 389f531d5..62c4c9c7a 100644
--- a/tests/jvm/src/test/resources/metacp.expect
+++ b/tests/jvm/src/test/resources/metacp.expect
@@ -3760,7 +3760,7 @@ Schema => SemanticDB v4
 Uri => types/Test.class
 Text => empty
 Language => Scala
-Symbols => 102 entries
+Symbols => 104 entries
 
 Symbols:
 types/Test. => final object Test extends AnyRef { +4 decls }
@@ -3973,13 +3973,19 @@ types/Test.C#typeRef4. => val method typeRef4: List[Int]
 types/Test.C#x. => val method x: p.X
   p => types/Test.C#p.
   X => types/P#X#
-types/Test.Literal. => final object Literal extends AnyRef { +12 decls }
+types/Test.Literal. => final object Literal extends AnyRef { +14 decls }
   AnyRef => scala/AnyRef#
 types/Test.Literal.bool. => final val method bool: true
 types/Test.Literal.char. => final val method char: 'a'
 types/Test.Literal.clazzOfInt. => final val method clazzOfInt: Class[Int]
   Class => java/lang/Class#
   Int => scala/Int#
+types/Test.Literal.clazzOfJStr. => final val method clazzOfJStr: Class[String]
+  Class => java/lang/Class#
+  String => java/lang/String#
+types/Test.Literal.clazzOfM. => final val method clazzOfM: Class[M]
+  Class => java/lang/Class#
+  M => types/Test.M#
 types/Test.Literal.clazzOfOption. => final val method clazzOfOption: Class[Option[Int]]
   Class => java/lang/Class#
   Option => scala/Option#
```

</details>

Reply:
This might be an unrelated issue then. Let's follow up with an issue and point to this exact place. Thanks for digging into it!